crates/walrus-sui/src/client/retry_client.rs

```diff
@@ -1001,19 +1112,43 @@ impl RetriableRpcClient {
         &self,
         sequence: u64,
     ) -> Result<CheckpointData, RetriableClientError> {
-        let primary_result = self.get_full_checkpoint_from_primary(sequence).await;
-
+        // First try with failover between endpoints.
+        let primary_result = self
+            .client
+            .with_failover(
+                |client| {
+                    Box::pin(async move {
+                        client
+                            .get_full_checkpoint(sequence)
+                            .await
+                            .map_err(RetriableClientError::from)
+                    })
+                },
+                self.request_timeout,
+            )
+            .await;
```

Comment:
This will quickly switch between endpoints upon any failure. I think we will probably want to retry first on the current endpoint to account for transient failures (network hiccups, overload, etc) before falling over. This also reduces unnecessary endpoint switching and gives more predictable behavior.  And for certain errors like "missing event" we most likely want to quickly fallback to the bucket instead of switching endpoints